### PR TITLE
Allow SignalToNoise to be zero

### PR DESCRIPTION
**Pull request by cquiroz**
Turns out there are edge cases where SignalToNoise can be 0
You can have an all noise picture for example and the ITC returns 0 in some edge cases (extremely faint stars, or odd component combinations)

### CODE REVIEW

File: modules/tests/shared/src/test/scala/lucuma/core/math/SignalToNoiseSuite.scala [hunk 1]
```diff
@@ -15,24 +18,30 @@ import lucuma.core.optics.Format
...
+    Gen.posNum[BigDecimal].map(refineV[NonNegative](_).getOrElse(throw new RuntimeException))
+  }
+
   checkAll("Order",                     OrderTests[SignalToNoise].order)
   checkAll("JSON Codec",                CodecTests[SignalToNoise].codec)
   checkAll("FromBigDecimalExact",       PrismTests(SignalToNoise.FromBigDecimalExact))
   checkAll("FromBigDecimalRounding",
     ValidSplitEpiTests(SignalToNoise.FromBigDecimalRounding).validSplitEpiWith(bigDecimalSignalToNoise))
-  checkAll("FromPosBigDecimalExact",    PrismTests(SignalToNoise.FromPosBigDecimalExact))
+  checkAll("FromPosBigDecimalExact",    PrismTests(SignalToNoise.FromNonNegBigDecimalExact))
```
Comment (swalker2m) [inline modules/tests/shared/src/test/scala/lucuma/core/math/SignalToNoiseSuite.scala:42]: I suppose we could update the label as well since it still reads "FromPosBigDecimal".

**Review by swalker2m — APPROVED** [review]
This seems good to me.  I can't remember now why I didn't include `0`.  We'll need to change a database constraint when we update the ODB for this.
